### ISSUE TITLE
Implement file invalidation based on new hash value sent by SURFsharekit

### ISSUE THREAD

**Issue by fako** [surfedushare/harvester#347]
Other Sharekit stuff we might do as well because we'll be testing Sharekit extensively anyway:
- [x] Give Sharekit access to harvester for webhook purposes
- [x] Use "owner" to extract provider for Sharekit in products
- [x] Main file should be the "important" file if available
- [x] Switch to edusources_acc and nppo_acc sets
- [x] Copyright should be set per file
- [x] Grab urlName to be title for links
- [x] Purge resources when invalidating tasks

**Comment by fako** [comment]
Had a look today, but hashes are not returned by the Sharekit API. So that makes it hard to use them. Will check with Dafne what happened here and put it on blocked for now @peterdubbeldsurf

**Comment by fako** [comment]
Furthermore. Implementing this is not as easy as picking up the hashes. We use the hashes in the file SRN (SURF resource name). To uniquely identify files. Products need to be able to generate these SRN's to do lookups correctly. And this is easy when using the URL based hash inside the SRN, but more difficult if the used hash is generated outside of the harvester.

Instead I think we should opt to pick up the hashes and then create SRN's based on the URL like before, but trigger background tasks based on url as well as hash. For most systems URL and hash would correlate 1:1, but for Sharekit it may happen that URL and SRN stays the same while the hash changes and this would trigger background tasks. To make things more complicated we'll generate a hash for Sharekit files when dealing with links, because Sharekit doesn't generate those (probably, haven't seen the API responses yet). All tests related to the logic described here is expected to break with the new API responses. So that's going to take some time as well.

TL; DR: this will be a bit more work, but it adds hours not days. Will need to be tested thoroughly, because we're changing most file tasks triggers for all sources.

**Comment by fako** [comment]
Can't give links importance in Sharekit
